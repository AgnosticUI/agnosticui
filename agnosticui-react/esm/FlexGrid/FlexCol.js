"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getColumnProps = getColumnProps;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createProps = _interopRequireDefault(require("./createProps"));

var _classNames = _interopRequireDefault(require("./classNames"));

var _types = require("./types");

var _excluded = ["tagName"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var propTypes = {
  /** xs - number of units when viewport is "extra small" */
  xs: _types.ColumnSizeType,

  /** sm - number of units when viewport is "small" */
  sm: _types.ColumnSizeType,

  /** md - number of units when viewport is "medium" */
  md: _types.ColumnSizeType,

  /** lg - number of units when viewport is "large" */
  lg: _types.ColumnSizeType,

  /** xl - number of units when viewport is "extra large" */
  xl: _types.ColumnSizeType,

  /** xsOffset - number of units to offset when viewport is "extra small" */
  xsOffset: _propTypes["default"].number,

  /** smOffset - number of units to offset when viewport is "small" */
  smOffset: _propTypes["default"].number,

  /** mdOffset - number of units to offset when viewport is "medium" */
  mdOffset: _propTypes["default"].number,

  /** lgOffset - number of units to offset when viewport is "large" */
  lgOffset: _propTypes["default"].number,

  /** xlOffset - number of units to offset when viewport is "extra large" */
  xlOffset: _propTypes["default"].number,

  /** first - Forces a column to appear first */
  first: _types.ViewportSizeType,

  /** last - Forces a column to appear last */
  last: _types.ViewportSizeType,

  /** className - additional custom class to use */
  className: _propTypes["default"].string,

  /** tagName - tag to use defaults to div */
  tagName: _propTypes["default"].string,

  /** children - react children */
  children: _propTypes["default"].node
};
var classMap = {
  xs: 'col-xs',
  sm: 'col-sm',
  md: 'col-md',
  lg: 'col-lg',
  xl: 'col-xl',
  xsOffset: 'col-xs-offset',
  smOffset: 'col-sm-offset',
  mdOffset: 'col-md-offset',
  lgOffset: 'col-lg-offset',
  xlOffset: 'col-xl-offset'
};

function isInteger(value) {
  return typeof value === 'number' && isFinite(value) && Math.floor(value) === value;
}

function getColClassNames(props) {
  var extraClasses = [];

  if (props.className) {
    extraClasses.push(props.className);
  }

  if (props.first) {
    extraClasses.push((0, _classNames["default"])('first-' + props.first));
  }

  if (props.last) {
    extraClasses.push((0, _classNames["default"])('last-' + props.last));
  }

  return Object.keys(props).filter(function (key) {
    return classMap[key];
  }).map(function (key) {
    return (0, _classNames["default"])(isInteger(props[key]) ? classMap[key] + '-' + props[key] : classMap[key]);
  }).concat(extraClasses);
}

function getColumnProps(props) {
  return (0, _createProps["default"])(propTypes, props, getColClassNames(props));
}

var FlexCol = function FlexCol(props) {
  var tagName = props.tagName,
      columnProps = _objectWithoutProperties(props, _excluded);

  return /*#__PURE__*/_react["default"].createElement(tagName || 'div', getColumnProps(columnProps));
};

FlexCol.propTypes = propTypes;
var _default = FlexCol;
exports["default"] = _default;