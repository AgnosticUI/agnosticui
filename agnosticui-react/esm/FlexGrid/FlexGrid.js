"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _createProps = _interopRequireDefault(require("./createProps"));

var _classNames = _interopRequireDefault(require("./classNames"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var propTypes = {
  fluid: _propTypes["default"].bool,
  className: _propTypes["default"].string,
  tagName: _propTypes["default"].string,
  children: _propTypes["default"].node
};
/**
 * FlexGrid is a flexbox based React component for doing layout. Under the hood it uses
 * a customized version of [flexboxgrid2](https://evgenyrodionov.github.io/flexboxgrid2)
 * that removes the magic gutters used by the original flexboxgrid2 css library. This
 * customized version (at `./FlexBoxGrid2Custom.css.scss`) is pulled in using CSS Modules.
 *
 * @public
 *
 * @param {string} className - in addition to sizing attributes, you can add custom CSS for the grid container
 * @param {string} tagName - what type of html tag to use for the container defaults to `<div>`
 * @param {ReactNode} children - children to render within the grid
 */

var FlexGrid = function FlexGrid(props) {
  var containerClass = (0, _classNames["default"])('flexgrid-container');
  var classNames = [props.className, containerClass];
  var gridPropsCreated = (0, _createProps["default"])(propTypes, props, classNames); // We're essentially taking the props passed in and:
  //   - rejecting any props passed in that are not in propTypes
  //   - adding classNames to our new react element below
  //   - allowing for a custom tagName or falling back to 'div'

  return /*#__PURE__*/_react["default"].createElement(props.tagName || 'div', gridPropsCreated);
};

FlexGrid.propTypes = propTypes;
var _default = FlexGrid;
exports["default"] = _default;