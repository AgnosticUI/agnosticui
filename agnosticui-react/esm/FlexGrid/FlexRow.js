"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getRowProps = getRowProps;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classNames = _interopRequireDefault(require("./classNames"));

var _createProps = _interopRequireDefault(require("./createProps"));

var _types = require("./types");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var rowKeys = ['start', 'center', 'end', 'top', 'middle', 'bottom', 'around', 'between'];
var propTypes = {
  reverse: _propTypes["default"].bool,
  start: _types.ViewportSizeType,
  center: _types.ViewportSizeType,
  end: _types.ViewportSizeType,
  top: _types.ViewportSizeType,
  middle: _types.ViewportSizeType,
  bottom: _types.ViewportSizeType,
  around: _types.ViewportSizeType,
  between: _types.ViewportSizeType,
  className: _propTypes["default"].string,
  tagName: _propTypes["default"].string,
  children: _propTypes["default"].node
};

function getRowClassNames(props) {
  var modifiers = [props.className, (0, _classNames["default"])('row')];

  for (var i = 0; i < rowKeys.length; ++i) {
    var key = rowKeys[i];
    var value = props[key];

    if (value) {
      modifiers.push((0, _classNames["default"])("".concat(key, "-").concat(value)));
    }
  }

  if (props.reverse) {
    modifiers.push((0, _classNames["default"])('reverse'));
  }

  return modifiers;
}

function getRowProps(props) {
  var rowPropsCreated = (0, _createProps["default"])(propTypes, props, getRowClassNames(props));
  return rowPropsCreated;
}

var FlexRow = function FlexRow(props) {
  return /*#__PURE__*/_react["default"].createElement(props.tagName || 'div', getRowProps(props));
};

FlexRow.propTypes = propTypes;
var _default = FlexRow;
exports["default"] = _default;