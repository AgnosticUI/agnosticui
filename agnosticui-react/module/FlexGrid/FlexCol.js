var _excluded = ["tagName"];

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import createProps from './createProps';
import getClass from './classNames';
import { ColumnSizeType, ViewportSizeType } from './types';
var propTypes = {
  /** xs - number of units when viewport is "extra small" */
  xs: ColumnSizeType,

  /** sm - number of units when viewport is "small" */
  sm: ColumnSizeType,

  /** md - number of units when viewport is "medium" */
  md: ColumnSizeType,

  /** lg - number of units when viewport is "large" */
  lg: ColumnSizeType,

  /** xl - number of units when viewport is "extra large" */
  xl: ColumnSizeType,

  /** xsOffset - number of units to offset when viewport is "extra small" */
  xsOffset: PropTypes.number,

  /** smOffset - number of units to offset when viewport is "small" */
  smOffset: PropTypes.number,

  /** mdOffset - number of units to offset when viewport is "medium" */
  mdOffset: PropTypes.number,

  /** lgOffset - number of units to offset when viewport is "large" */
  lgOffset: PropTypes.number,

  /** xlOffset - number of units to offset when viewport is "extra large" */
  xlOffset: PropTypes.number,

  /** first - Forces a column to appear first */
  first: ViewportSizeType,

  /** last - Forces a column to appear last */
  last: ViewportSizeType,

  /** className - additional custom class to use */
  className: PropTypes.string,

  /** tagName - tag to use defaults to div */
  tagName: PropTypes.string,

  /** children - react children */
  children: PropTypes.node
};
var classMap = {
  xs: 'col-xs',
  sm: 'col-sm',
  md: 'col-md',
  lg: 'col-lg',
  xl: 'col-xl',
  xsOffset: 'col-xs-offset',
  smOffset: 'col-sm-offset',
  mdOffset: 'col-md-offset',
  lgOffset: 'col-lg-offset',
  xlOffset: 'col-xl-offset'
};

function isInteger(value) {
  return typeof value === 'number' && isFinite(value) && Math.floor(value) === value;
}

function getColClassNames(props) {
  var extraClasses = [];

  if (props.className) {
    extraClasses.push(props.className);
  }

  if (props.first) {
    extraClasses.push(getClass('first-' + props.first));
  }

  if (props.last) {
    extraClasses.push(getClass('last-' + props.last));
  }

  return Object.keys(props).filter(function (key) {
    return classMap[key];
  }).map(function (key) {
    return getClass(isInteger(props[key]) ? classMap[key] + '-' + props[key] : classMap[key]);
  }).concat(extraClasses);
}

export function getColumnProps(props) {
  return createProps(propTypes, props, getColClassNames(props));
}

var FlexCol = function FlexCol(props) {
  var tagName = props.tagName,
      columnProps = _objectWithoutProperties(props, _excluded);

  return /*#__PURE__*/React.createElement(tagName || 'div', getColumnProps(columnProps));
};

FlexCol.propTypes = propTypes;
export default FlexCol;