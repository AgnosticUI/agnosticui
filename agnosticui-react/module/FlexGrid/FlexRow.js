import React from 'react';
import PropTypes from 'prop-types';
import getClass from './classNames';
import createProps from './createProps';
import { ViewportSizeType } from './types';
var rowKeys = ['start', 'center', 'end', 'top', 'middle', 'bottom', 'around', 'between'];
var propTypes = {
  reverse: PropTypes.bool,
  start: ViewportSizeType,
  center: ViewportSizeType,
  end: ViewportSizeType,
  top: ViewportSizeType,
  middle: ViewportSizeType,
  bottom: ViewportSizeType,
  around: ViewportSizeType,
  between: ViewportSizeType,
  className: PropTypes.string,
  tagName: PropTypes.string,
  children: PropTypes.node
};

function getRowClassNames(props) {
  var modifiers = [props.className, getClass('row')];

  for (var i = 0; i < rowKeys.length; ++i) {
    var key = rowKeys[i];
    var value = props[key];

    if (value) {
      modifiers.push(getClass("".concat(key, "-").concat(value)));
    }
  }

  if (props.reverse) {
    modifiers.push(getClass('reverse'));
  }

  return modifiers;
}

export function getRowProps(props) {
  var rowPropsCreated = createProps(propTypes, props, getRowClassNames(props));
  return rowPropsCreated;
}

var FlexRow = function FlexRow(props) {
  return /*#__PURE__*/React.createElement(props.tagName || 'div', getRowProps(props));
};

FlexRow.propTypes = propTypes;
export default FlexRow;