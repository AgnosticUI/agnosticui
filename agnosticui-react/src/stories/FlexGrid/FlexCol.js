import React from 'react';
import PropTypes from 'prop-types';
import createProps from './createProps';
import getClass from './classNames';
import { ColumnSizeType, ViewportSizeType } from './types';

const propTypes = {
  /** xs - number of units when viewport is "extra small" */
  xs: ColumnSizeType,
  /** sm - number of units when viewport is "small" */
  sm: ColumnSizeType,
  /** md - number of units when viewport is "medium" */
  md: ColumnSizeType,
  /** lg - number of units when viewport is "large" */
  lg: ColumnSizeType,
  /** xl - number of units when viewport is "extra large" */
  xl: ColumnSizeType,
  /** xsOffset - number of units to offset when viewport is "extra small" */
  xsOffset: PropTypes.number,
  /** smOffset - number of units to offset when viewport is "small" */
  smOffset: PropTypes.number,
  /** mdOffset - number of units to offset when viewport is "medium" */
  mdOffset: PropTypes.number,
  /** lgOffset - number of units to offset when viewport is "large" */
  lgOffset: PropTypes.number,
  /** xlOffset - number of units to offset when viewport is "extra large" */
  xlOffset: PropTypes.number,
  /** first - Forces a column to appear first */
  first: ViewportSizeType,
  /** last - Forces a column to appear last */
  last: ViewportSizeType,
  /** className - additional custom class to use */
  className: PropTypes.string,
  /** tagName - tag to use defaults to div */
  tagName: PropTypes.string,
  /** children - react children */
  children: PropTypes.node,
};

const classMap = {
  xs: 'col-xs',
  sm: 'col-sm',
  md: 'col-md',
  lg: 'col-lg',
  xl: 'col-xl',
  xsOffset: 'col-xs-offset',
  smOffset: 'col-sm-offset',
  mdOffset: 'col-md-offset',
  lgOffset: 'col-lg-offset',
  xlOffset: 'col-xl-offset',
};

function isInteger(value) {
  return typeof value === 'number' && isFinite(value) && Math.floor(value) === value;
}

function getColClassNames(props) {
  const extraClasses = [];

  if (props.className) {
    extraClasses.push(props.className);
  }

  if (props.first) {
    extraClasses.push(getClass('first-' + props.first));
  }

  if (props.last) {
    extraClasses.push(getClass('last-' + props.last));
  }

  return Object.keys(props)
    .filter((key) => classMap[key])
    .map((key) =>
      getClass(isInteger(props[key]) ? classMap[key] + '-' + props[key] : classMap[key])
    )
    .concat(extraClasses);
}

export function getColumnProps(props) {
  return createProps(propTypes, props, getColClassNames(props));
}

const FlexCol = (props) => {
  const { tagName, ...columnProps } = props;
  return React.createElement(tagName || 'div', getColumnProps(columnProps));
};

FlexCol.propTypes = propTypes;

export default FlexCol;
