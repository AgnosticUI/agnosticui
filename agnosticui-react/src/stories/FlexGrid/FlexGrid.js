import React from 'react'
import PropTypes from 'prop-types'
import createProps from './createProps'
import getClass from './classNames'

const propTypes = {
  fluid: PropTypes.bool,
  className: PropTypes.string,
  tagName: PropTypes.string,
  children: PropTypes.node,
}

/**
 * FlexGrid is a flexbox based React component for doing layout. Under the hood it uses
 * a customized version of [flexboxgrid2](https://evgenyrodionov.github.io/flexboxgrid2)
 * that removes the magic gutters used by the original flexboxgrid2 css library. This
 * customized version (at `./FlexBoxGrid2Custom.css.scss`) is pulled in using CSS Modules.
 *
 * @public
 *
 * @param {string} className - in addition to sizing attributes, you can add custom CSS for the grid container
 * @param {string} tagName - what type of html tag to use for the container defaults to `<div>`
 * @param {ReactNode} children - children to render within the grid
 */
const FlexGrid = (props) => {
  const containerClass = getClass('flexgrid-container')
  const classNames = [props.className, containerClass]
  const gridPropsCreated = createProps(propTypes, props, classNames)

  // We're essentially taking the props passed in and:
  //   - rejecting any props passed in that are not in propTypes
  //   - adding classNames to our new react element below
  //   - allowing for a custom tagName or falling back to 'div'
  return React.createElement(
    props.tagName || 'div',
    gridPropsCreated
  )
}

FlexGrid.propTypes = propTypes

export default FlexGrid;
