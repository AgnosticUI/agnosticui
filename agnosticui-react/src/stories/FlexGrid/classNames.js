import styles from './FlexBoxGrid2Custom.css'

export default function getClass(className) {
  return styles && styles[className] ? styles[className] : className
}
