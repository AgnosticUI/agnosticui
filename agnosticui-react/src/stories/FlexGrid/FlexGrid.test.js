import React from 'react'
import FlexGrid from './FlexGrid';
import FlexRow from './FlexRow';
import FlexCol from './FlexCol';
import styles from './FlexGridExample.module.css'
import renderer from 'react-test-renderer'

const Field = () => <p className={styles.Field}>Field</p>

describe('FlexGrid', () => {
  describe('API', () => {
    test('exports properly', () => {
      expect(FlexGrid).toBeDefined()
      expect(FlexRow).toBeDefined()
      expect(FlexCol).toBeDefined()
    })
  })

  describe('rendering component', () => {
    test('default', () => {
      const tree = renderer
        .create(
          <FlexGrid>
            <FlexRow className={styles.DemoCol}>
              <FlexCol xs={12} sm={4} lg={6}>
                <Field />
              </FlexCol>
              <FlexCol xs={12} sm={4} lg={6}>
                <Field />
              </FlexCol>
            </FlexRow>
          </FlexGrid>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
