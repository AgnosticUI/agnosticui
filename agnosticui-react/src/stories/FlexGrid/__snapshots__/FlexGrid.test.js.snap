// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexGrid rendering component default 1`] = `
<div
  className="flexgrid-container"
>
  <div
    className="DemoCol row"
  >
    <div
      className="col-xs-12 col-sm-4 col-lg-6"
    >
      <p
        className="Field"
      >
        Field
      </p>
    </div>
    <div
      className="col-xs-12 col-sm-4 col-lg-6"
    >
      <p
        className="Field"
      >
        Field
      </p>
    </div>
  </div>
</div>
`;
